platform/ui/src/contextProviders/ContextMenuProvider.js

```diff
@@ -0,0 +1,136 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useCallback,
+} from 'react';
+import PropTypes from 'prop-types';
+
+import ToolContextMenu from '../../../viewer/src/connectedComponents/ToolContextMenu';
+
+import { useDialog } from './DialogProvider';
+import { useLabellingFlow } from './LabellingFlowProvider';
+
+const ContextMenuContext = createContext(null);
+const { Provider } = ContextMenuContext;
+
+export const useContextMenu = () => useContext(ContextMenuContext);
+
+const ContextMenuProvider = ({ children, service }) => {
+  const { create, dismiss } = useDialog();
```

Comment:
Okay, so it is using `useDialog` under the hood :+1:
We're just exposing it as a named service, and this separate context provider keeps the state separate 🤔 

Interesting stuff to think about.

Reply:
Yeah the providers itself follows the same context rules as other components inside app, they can access different contexts to leverage the existent generic components, this way they can extend themselves.